main.tf

```diff
@@ -41,6 +40,13 @@ locals {
         secret_name    = secret_name
         secret_data    = secret_data
   }]])
+
+  firewall_name = format("%s-%s", "ingress-allow", var.name_suffix)
+  istio_ports = [
+    "10250", "443", "15017", # see https://istio.io/latest/docs/setup/platform-setup/gke/
+    "8080", "15000",         # see https://kiali.io/documentation/latest/installation-guide/#_google_cloud_private_cluster_requirements
+  ]
+  firewall_ports = toset(concat(var.additional_firewall_ports, length(toset(var.istio_ip_names)) > 0 ? local.istio_ports : []))
```

Comment:
there are two things to take care here, one is istio is enabled, second is additional_ingress_port is set

Reply:
That's ok. If istio is enabled, then only we populate `local.istio_ports`. Otherwise, `local.istio_ports` can be `[]`. So that decision is taken care of in the context `local.istio_ports` itself.

Here, we just concat it. 

Hence, If `local.istio_ports` is empty, then here it will be empty + var.additional_firewall_ports. If `local.istio_ports` has values, then here it will be some values + var.additional_firewall_ports